src/object.js

```diff
@@ -143,6 +145,9 @@ inherits(ObjectSchema, MixedSchema, {
           return value;
         }
 
+        from = originalValue
+          ? from
```

Comment:
should probably not overload the variable here so children get a fresh, new array, incase the mutate it it won't break anything down the line?

Reply:
Would this be correct?

```
from = originalValue ? [...from] : ...
```

Reply:
ya